pyvo/mivot/viewer/mivot_instance.py

```diff
@@ -0,0 +1,173 @@
+        row (astropy.table.row.Row): The new data row.
+        ref (str, optional):The reference of the data row, default is None.
+        """
+        for key, value in vars(self).items():
+            if isinstance(value, list):
+                for item in value:
+                    item.update(row=row)
+            elif isinstance(value, MivotInstance):
+                if isinstance(vars(value), dict):
+                    if 'value' not in vars(value):
+                        value.update(row=row)
+                    if 'value' in vars(value):
+                        value.update(row=row, ref=getattr(value, 'ref'))
+            else:
+                if key == 'value':
+                    if ref is not None and ref != 'null':
+                        setattr(self, self._remove_model_name(key),
+                                MivotUtils.cast_type_value(row[ref], getattr(self, 'dmtype')))
+
+    @staticmethod
+    def _remove_model_name(value, role_instance=False):
+        """
+        Remove the model name before each colon ":" as well as the type of the object before each point ".".
+        If it is an INSTANCE of INSTANCEs, the dmrole represented as the key needs to keep his type object.
+        In this case (`role_instance=True`), we just replace the point "." With an underscore "_".
+        Parameters
+        ----------
+        value (str): The string to process.
+        role_instance (bool, optional): If True, keeps the type object for dmroles representing
+                                        an INSTANCE of INSTANCEs. Default is False.
+        """
+        if isinstance(value, str):
+            # We first find the model_name before the colon
+            index_underscore = value.find(":")
+            if index_underscore != -1:
+                # Then we find the object type before the point
+                next_index_underscore = value.find(".", index_underscore + 1)
+                if next_index_underscore != -1 and not role_instance:
+                    value_after_underscore = value[next_index_underscore + 1:]
+                else:
+                    value_after_underscore = (value[index_underscore + 1:]
+                                              .replace(':', '_').replace('.', '_'))
+                return value_after_underscore
+            return value  # Returns unmodified string if "_" wasn't found
+        else:
+            return value
+
+    def _is_leaf(self, **kwargs):
+        """
+        Check if the dictionary is an ATTRIBUTE.
+        Parameters
+        ----------
+        **kwargs (dict): The dictionary to check.
+        Returns
+        -------
+        bool: True if the dictionary is an ATTRIBUTE, False otherwise.
+        """
+        if isinstance(kwargs, dict):
+            for _, value in kwargs.items():
+                if isinstance(value, dict):
+                    return False
+        return True
+
+    def display_class_dict(self, obj, classkey=None):
```

Comment:
I think a `__repr__` or `__str__` method would be really useful for users to explore a `MivotInstance` result.

Reply:
Added to `MivotInstance` 2 methods generating class dicts:
- `dict():` return a dictionary where leaves only contain `@value` and `@unit`
- `hk_dict()`: return a dictionary where leaves also contain `@ref,` `@dmtype` and `@astropy_unit`

The `__repr__` method builds a pretty string from the `dic()` output